src/Schedgen2/mpi_colls.py

```diff
@@ -56,103 +27,185 @@ def dissemination(comm_size, datasize, tag):
     return comm
 
 
-def recdoub_allreduce(comm, comm_size, datasize, base_tag, ctd=0):
-    num_steps = int(log2(comm_size))
-    for rank in range(0, comm_size):
-        # Reduce-scatter
-        sources = [rank ^ (2**i) for i in range(num_steps)]
-        destinations = sources
-        data_sizes_receive = [datasize // (2**i) for i in range(1, num_steps + 1)]
-        data_sizes_send = data_sizes_receive
-        dependency = iterative_send_recv(
-            comm,
-            rank,
-            sources,
-            destinations,
-            data_sizes_receive,
-            data_sizes_send,
-            base_tag,
-            compute_time_dependency=ctd,
-        )
+def incast(
```

Comment:
same as below, MPI_Incast does not exist, remove

Reply:
will move to a new file